pkg/engine/adapters/rclient.go

```diff
@@ -33,6 +33,10 @@ func (a *rclientAdapter) ForRef(ctx context.Context, ref string) (*engineapi.Ima
 	if err != nil {
 		return nil, fmt.Errorf("failed to resolve image reference: %s, error: %v", ref, err)
 	}
+	imageIndex, err := desc.ImageIndex()
```

Comment:
ImageIndex only returns an index if the image is a multiplatform image, it returns an error if the image supports only one platfrom.
For example, a single platfrom image like `ghcr.io/kyverno/test-verify-image:signed` returns this error: `unexpected media type for ImageIndex(): application/vnd.docker.distribution.manifest.v2+json; call Image() instead`

Reply:
In the case of a single-platform image, Should Kyverno produce a warning and return `nil` for this field?